monkestation/code/modules/clothing/bodycams/Bodycamcode.dm

```diff
@@ -0,0 +1,145 @@
+
+/obj/item/clothing/suit/armor/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/bodycamera_holder)
+
+/obj/item/clothing/suit/hooded/wintercoat/security/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/bodycamera_holder)
+/**
+ * The bodycamera
+ *
+ * This is the item that gets installed into items that have the bodycamera_holder element
+ */
+/obj/item/bodycam_upgrade
+	name = "\improper body camera"
+	icon = 'monkestation/code/modules/clothing/bodycams/bodycamera.dmi'
+	icon_state = "bodycamera"
+	desc = "An armor vest upgrade, there's an instructions tag if you look a little closer..."
+	///The camera itself.
+	var/obj/machinery/camera/builtin_bodycamera
+
+/obj/item/bodycam_upgrade/examine_more(mob/user)
+	. = ..()
+	. += list(span_notice("Use [src] on any valid vest to quickly install."))
+	. += list(span_notice("Use a [span_bold("screwdriver")] to remove it."))
+	. += list(span_notice("While equipped, use your ID card on the vest to activate/deactivate the camera."))
+	. += list(span_notice("Unequipping the vest will immediately deactivate the camera."))
+
+/obj/item/bodycam_upgrade/Destroy()
+	if(builtin_bodycamera)
+		turn_off()
+	return ..()
+
+/obj/item/bodycam_upgrade/proc/is_on()
+	if(isnull(builtin_bodycamera))
+		return FALSE
+	return TRUE
+
+/obj/item/bodycam_upgrade/proc/turn_on(mob/user, obj/item/card/id/id_card)
+	builtin_bodycamera = new(user)
+	builtin_bodycamera.internal_light = FALSE
+	builtin_bodycamera.network = list("ss13")
+	builtin_bodycamera.c_tag = "-Body Camera: [(id_card.registered_name)] ([id_card.assignment])"
+	playsound(loc, 'sound/machines/beep.ogg', get_clamped_volume(), TRUE, -1)
+	if(user)
+		user.balloon_alert(user, "bodycamera activated.")
+
+/obj/item/bodycam_upgrade/proc/turn_off(mob/user)
+	if(user)
+		user.balloon_alert(user, "bodycamera deactivated.")
+	playsound(loc, 'sound/machines/beep.ogg', get_clamped_volume(), TRUE, -1)
+	QDEL_NULL(builtin_bodycamera)
+
+
+/**
+ * Bodycamera component
```

Comment:
You should move this component to a new file, it would be like ``bodycamera_item.dm`` and ``bodycamera_component.dm``